lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -0,0 +1,283 @@
+          </NxDrawer.Header>
+        )
+      });
+
+      const closeButton = screen.getByRole('button', { hidden: true, name: /close/i });
+
+      expect(closeButton).toBeInTheDocument();
+      expect(closeButton).toHaveClass('nx-drawer-header__close-button');
+    });
+
+    it('renders title, subtitle, description with the correct tags', function() {
+      quickRender({
+        children: (
+          <NxDrawer.Header>
+            <NxDrawer.HeaderTitle>Title</NxDrawer.HeaderTitle>
+            <NxDrawer.HeaderSubtitle>Subtitle</NxDrawer.HeaderSubtitle>
+            <NxDrawer.HeaderDescription>Description</NxDrawer.HeaderDescription>
+          </NxDrawer.Header>
+        )
+      });
+
+      const headerTitle = screen.getByRole('heading', { name: 'Title', hidden: true });
+      const headerSubtitle = screen.getByRole('heading', { name: 'Subtitle', hidden: true });
+      const headerDescription = screen.getByText('Description');
+
+      expect(headerTitle).toBeInTheDocument();
+      expect(headerTitle.nodeName).toBe('H2');
+
+      expect(headerSubtitle).toBeInTheDocument();
+      expect(headerSubtitle.nodeName).toBe('H3');
+
+      expect(headerDescription).toBeInTheDocument();
+      expect(headerDescription.nodeName).toBe('P');
+    });
+
+    it('renders title, subtitle, description with the correct classnames', function() {
+      quickRender({
+        children: (
+          <NxDrawer.Header>
+            <NxDrawer.HeaderTitle>Title</NxDrawer.HeaderTitle>
+            <NxDrawer.HeaderSubtitle>Subtitle</NxDrawer.HeaderSubtitle>
+            <NxDrawer.HeaderDescription>Description</NxDrawer.HeaderDescription>
+          </NxDrawer.Header>
+        )
+      });
+
+      const headerTitle = screen.getByRole('heading', { name: 'Title', hidden: true });
+      const headerSubtitle = screen.getByRole('heading', { name: 'Subtitle', hidden: true });
+      const headerDescription = screen.getByText('Description');
+
+      expect(headerTitle).toHaveClass('nx-drawer-header__title');
+      expect(headerSubtitle).toHaveClass('nx-drawer-header__subtitle');
+      expect(headerDescription).toHaveClass('nx-drawer-header__description');
+    });
+
+    it('executes onClose when header close button is clicked', async function() {
+      const user = userEvent.setup();
+      const mockOnClose = jest.fn();
+
+      quickRender({
+        onClose: mockOnClose,
+        children: (
+          <NxDrawer.Header>
+            <NxDrawer.HeaderTitle>Hello</NxDrawer.HeaderTitle>
+          </NxDrawer.Header>
+        )
+      });
+
+      const closeButton = screen.getByRole('button', { hidden: true, name: 'Close' });
+
+      expect(closeButton).toBeInTheDocument();
+      expect(mockOnClose).not.toHaveBeenCalled();
+
+      await act(async () => {
+        await user.click(closeButton);
+      });
+
+      expect(mockOnClose).toHaveBeenCalled();
+      expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
```

Comment:
I don't understand why this is true.  the `mockOnClose` doesn't actually do anything

Reply:
> I don't understand why this is true. the `mockOnClose` doesn't actually do anything

Ops it's because `{hidden:true}` was not specified in the query.
I simplified the whole `it()` block here: fcbdb63270a00c70a4347993621d7e99f7a87969